tsfc/driver.py

```diff
@@ -60,21 +53,17 @@ def compile_form(form, prefix="form", parameters=None):
     return kernels
 
 
-def compile_integral(integral_data, form_data, prefix, parameters):
+def compile_integral(integral_data, form_data, prefix, parameters,
+                     backend="pyop2"):
     """Compiles a UFL integral into an assembly kernel.
 
     :arg integral_data: UFL integral data
     :arg form_data: UFL form data
     :arg prefix: kernel name will start with this string
     :arg parameters: parameters object
+    :arg backend: output format
     :returns: a kernel, or None if the integral simplifies to zero
     """
-    # Remove these here, they're handled below.
```

Comment:
Can you explain your tinkering with the parameters in this function? In Firedrake, the global parameters (coming from the `parameters` argument of this function) generally override the parameters specified as integral metadata, with these exceptions here: if `quadrature_degree` or `quadrature_rule` are set to `auto` then they don't override the per-integral specified quadrature degree (if any). Since the `auto` values are no longer removed in this patch, they will always override the per-integral degree in [line 177](https://github.com/firedrakeproject/tsfc/pull/54/files#diff-4906bef44a1f7f6a1cbf093c68d66c37R177), causing the test failure with `f*dx(degree=1)` for example.

Can you explain what is the expected behaviour in FFC and what values are coming from the `parameters` argument?